language_service/src/completion.rs

```diff
@@ -121,6 +131,16 @@ fn expected_word_kinds(
     }
 }
 
+fn in_comment(source_contents: &str, cursor_offset: u32) -> bool {
+    // find the last newline before the cursor
+    let last_line_start = source_contents[..cursor_offset as usize]
+        .rfind('\n')
+        .unwrap_or(0);
+    // find the last comment start before the cursor
+    let last_comment_start = source_contents[last_line_start..cursor_offset as usize].rfind("//");
```

Comment:
Probably not worth worrying about, but I assume `//` anywhere in the line will make it think it is in a comment (i.e. even if those chars are in a string literal - which seems like a low enough probability edge case not to woryy about).

Reply:
Yes, it shouldn't be that common. I think it is worth it to commit to this now and file an issue for an improvement.

Reply:
Per side discussion - this is a valid concern, but requires a bit larger fix. Instead of patching up this case, I'd rather spend a bit more time to look at how we fix the root cause (in the lexer itself). I'll open an issue for it.